src/async_wrap.cc

```diff
@@ -198,37 +200,110 @@ class PromiseWrap : public AsyncWrap {
     MakeWeak();
   }
 
+  PromiseWrap(Environment* env, Local<Object> object, double asyncId,
+    double triggerAsyncId)
+      : AsyncWrap(env, object, PROVIDER_PROMISE, asyncId, triggerAsyncId) {
+    MakeWeak();
+  }
+
   SET_NO_MEMORY_INFO()
   SET_MEMORY_INFO_NAME(PromiseWrap)
   SET_SELF_SIZE(PromiseWrap)
 
   static PromiseWrap* New(Environment* env,
                           Local<Promise> promise,
                           PromiseWrap* parent_wrap,
+                          bool isInit,
                           bool silent);
-  static void getIsChainedPromise(Local<String> property,
-                                  const PropertyCallbackInfo<Value>& info);
+  static void getAsyncId(Local<Name> property,
+                         const PropertyCallbackInfo<Value>& args);
+  static void getTriggerAsyncId(Local<Name> property,
+                                const PropertyCallbackInfo<Value>& args);
+
+  static void Initialize(Environment* env);
 };
 
 PromiseWrap* PromiseWrap::New(Environment* env,
                               Local<Promise> promise,
                               PromiseWrap* parent_wrap,
+                              bool isInit,
```

Comment:
This argument always equals to `!silent`, so maybe it makes sense to keep only one of them?

Reply:
I just did for the parent_wrap case, but looking at it closer that's not actually even _needed_ anymore due to dropping `isChainedPromise`. 🤔

Reply:
Oh, actually it's still needed for the `DefaultTriggerAsyncIdScope`. If I just use the `silent` property than it will do the checks any time it's wrapping a parent promise, because it is _always_ silent.